tools/python/util/convert_onnx_models_to_ort.py

```diff
@@ -78,21 +78,16 @@ def _convert(model_path_or_dir: pathlib.Path, optimization_level_str: str, use_n
     if len(models) == 0:
         raise ValueError("No .onnx files were found in '{}'".format(model_path_or_dir))
 
-    providers = ['CPUExecutionProvider']
-    if use_nnapi:
-        # providers are priority based, so register NNAPI first
-        providers.insert(0, 'NnapiExecutionProvider')
-    if use_coreml:
-        # providers are priority based, so register CoreML first
-        providers.insert(0, 'CoreMLExecutionProvider')
-
     # if the optimization level is 'all' we manually exclude the NCHWc transformer. It's not applicable to ARM
     # devices, and creates a device specific model which won't run on all hardware.
     # If someone really really really wants to run it they could manually create an optimized onnx model first,
     # or they could comment out this code.
-    optimizer_filter = None
+    optimizer_filter = set()
     if optimization_level == ort.GraphOptimizationLevel.ORT_ENABLE_ALL and target_platform != 'amd64':
-        optimizer_filter = ['NchwcTransformer']
+        optimizer_filter.add('NchwcTransformer')
+    if "OpenCLExecutionProvider" in providers:
+        optimizer_filter.add('NchwcTransformer')
```

Comment:
What's the reason for this? The NCHWc transformer is amd64 specific afaik. How is it relevent to OpenCL?

Reply:
Removed.

My bad, forgot to check the list in `graph_transformer_utils.cc`.